collector/src/main/java/com/bitmovin/analytics/retryBackend/RetrySample.kt

```diff
@@ -0,0 +1,16 @@
+package com.bitmovin.analytics.retryBackend
+
+import com.bitmovin.analytics.data.AdEventData
+import com.bitmovin.analytics.data.EventData
+import java.util.Date
+
+class RetrySample(val eventData: EventData?, val adEventData: AdEventData?, var totalTime: Int, var scheduledTime: Date, var retry: Int) : Comparable<RetrySample> {
+    override fun compareTo(other: RetrySample): Int {
+        return when {
+            (this == null && other == null) -> 0
```

Comment:
`this` can never be null here.
Also, as `scheduledTime` is not nullable, it would probably be enough to have 
`return this.scheduledTime.compareTo(other?.scheduledTime)`.